OpenRA.Game/GameRules/ActorInfo.cs

```diff
@@ -184,8 +184,13 @@ static IEnumerable<Type> PrerequisitesOf(ITraitInfo info)
 					i.Name.Replace("Init", ""), i));
 		}
 
-		public bool HasTraitInfo<T>() where T : ITraitInfo { return Traits.Contains<T>(); }
-		public T TraitInfo<T>() where T : ITraitInfo { return Traits.Get<T>(); }
-		public T TraitInfoOrDefault<T>() where T : ITraitInfo { return Traits.GetOrDefault<T>(); }
+		public bool HasTraitInfo<T>() where T : ITraitInfo { return traits.Contains<T>(); }
+		public T TraitInfo<T>() where T : ITraitInfo { return traits.Get<T>(); }
+		public T TraitInfoOrDefault<T>() where T : ITraitInfo { return traits.GetOrDefault<T>(); }
+		public IEnumerable<ITraitInfo> TraitInfos() { return traits.WithInterface<ITraitInfo>(); }
```

Comment:
In some places you use `TraitInfos()` and in others you use `TraitInfos<ITraitInfo>()`.  I think it would be clearer to use `TraitInfos<ITraitInfo>()` everywhere and then remove this version.